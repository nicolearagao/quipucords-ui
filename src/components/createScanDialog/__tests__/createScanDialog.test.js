import React from 'react';
import { CreateScanDialog } from '../createScanDialog';

describe('CreateScanDialog Component', () => {
  it('should render a basic component', () => {
    const props = {
      show: true,
      sources: [{ name: 'test name' }]
    };

    const component = renderComponent(<CreateScanDialog {...props} />);
    expect(component.screen.render()).toMatchSnapshot('basic');
  });

  it('should render nothing if sources are not provided', () => {
    const props = {
      show: true,
      sources: []
    };

    const component = renderComponent(<CreateScanDialog {...props} />);
    expect(component.screen.render()).toMatchSnapshot('empty');
  });

  it('should render a component, pending', () => {
    const props = {
      pending: true,
      show: true,
      sources: [{ name: 'test name' }]
    };

    const component = renderComponent(<CreateScanDialog {...props} />);
    expect(component.screen.render().querySelector('.pf-c-modal-box__body')).toMatchSnapshot('pending');
  });

  it('should handle multiple error responses', () => {
    const props = {
      show: true,
      sources: [{ name: 'test name' }],
      error: true,
      errorMessage: 'lorem ipsum'
    };

    const component = renderComponent(<CreateScanDialog {...props} />);
    expect(component.screen.getByLabelText('Danger Alert')).toMatchSnapshot('basic error');

    const componentNamedError = component.setProps({ submitErrorMessages: { scanName: 'dolor sit' } });
    expect(componentNamedError.screen.render().querySelectorAll('[id*="scanName"]')).toMatchSnapshot('named error');
  });

  it('should correctly validate data', () => {
    const props = {
      show: true,
      sources: [{ name: 'test name' }]
    };

    const component = renderComponent(<CreateScanDialog {...props} />);

    expect(
      component.instance.onValidateForm({
        values: {
          scanName: '',
          scanDirectories: ['/ipsum']
        },
        checked: {
          jbossEap: false
        }
      })
    ).toMatchSnapshot('form invalid');

    expect(
      component.instance.onValidateForm({
        values: {
          scanName: 'lorem',
          scanDirectories: ['/ipsum']
        },
        checked: {
          jbossEap: true
        }
      })
    ).toMatchSnapshot('form valid');
  });
});
