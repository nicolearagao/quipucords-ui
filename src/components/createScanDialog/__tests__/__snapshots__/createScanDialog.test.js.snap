// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateScanDialog Component should correctly validate data: form invalid 1`] = `
{
  "scanDirectories": true,
  "scanName": true,
}
`;

exports[`CreateScanDialog Component should correctly validate data: form valid 1`] = `
{
  "scanDirectories": false,
  "scanName": false,
}
`;

exports[`CreateScanDialog Component should handle multiple error responses: basic error 1`] = `
<div
  aria-label="Danger Alert"
  class="pf-c-alert pf-m-inline pf-m-danger"
  data-ouia-component-id="OUIA-Generated-Alert-danger-1"
  data-ouia-component-type="PF4/Alert"
  data-ouia-safe="true"
>
  <div
    class="pf-c-alert__icon"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      height="1em"
      role="img"
      style="vertical-align: -0.125em;"
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
      />
    </svg>
  </div>
  <h4
    class="pf-c-alert__title"
  >
    <span
      class="pf-u-screen-reader"
    >
      Danger alert:
    </span>
    t(form-dialog.label, {"context":"error"})
  </h4>
  <div
    class="pf-c-alert__description"
  >
    lorem ipsum
  </div>
</div>
`;

exports[`CreateScanDialog Component should handle multiple error responses: named error 1`] = `
NodeList [
  <input
    aria-invalid="false"
    class="pf-c-form-control quipucords-form__text-input "
    context\\":\\"placeholder\\"})"=""
    data-ouia-component-id="OUIA-Generated-TextInputBase-3"
    data-ouia-component-type="PF4/TextInput"
    data-ouia-safe="true"
    id="scanName"
    maxlength="256"
    name="scanName"
    placeholder="t(form-dialog.label_scan-name, {\\"
    type="text"
    value=""
  />,
  <div
    aria-live="polite"
    class="pf-c-form__helper-text pf-m-error"
    id="scanName-helper"
  >
     dolor sit 
  </div>,
]
`;

exports[`CreateScanDialog Component should render a basic component: basic 1`] = `
<screen>
  <div
    class="quipucords-modal quipucords-modal__rcue-width pf-c-backdrop__open"
  >
    <div>
      <div
        class="pf-c-backdrop"
      >
        <div
          class="pf-l-bullseye"
        >
          <div
            aria-describedby="pf-modal-part-2"
            aria-label="t(modal.aria-label-default)"
            aria-labelledby="pf-modal-part-0"
            aria-modal="true"
            class="pf-c-modal-box pf-m-align-top"
            data-ouia-component-id="OUIA-Generated-Modal-1"
            data-ouia-component-type="PF4/ModalContent"
            data-ouia-safe="true"
            id="pf-modal-part-0"
            role="dialog"
            style="--pf-c-modal-box--m-align-top--spacer: 5%;"
          >
            <button
              aria-disabled="false"
              aria-label="Close"
              class="pf-c-button pf-m-plain"
              data-ouia-component-id="OUIA-Generated-Modal-1-ModalBoxCloseButton"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
            <header
              class="pf-c-modal-box__header"
            >
              <h4
                class="pf-c-title pf-m-md"
                data-ouia-component-id="OUIA-Generated-Title-1"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                 t(form-dialog.title, {"context":"create-scan"}) 
              </h4>
            </header>
            <div
              class="pf-c-modal-box__body"
              id="pf-modal-part-2"
            >
              <form
                class="pf-c-form pf-m-horizontal"
                novalidate=""
              >
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="scanName"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                         t(form-dialog.label, {"context":"scan-name"}) 
                      </span>
                    </label>
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      class="pf-c-form-control quipucords-form__text-input "
                      context\\":\\"placeholder\\"})"=""
                      data-ouia-component-id="OUIA-Generated-TextInputBase-1"
                      data-ouia-component-type="PF4/TextInput"
                      data-ouia-safe="true"
                      id="scanName"
                      maxlength="256"
                      name="scanName"
                      placeholder="t(form-dialog.label_scan-name, {\\"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="displayScanSources"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                         t(form-dialog.label, {"context":"sources"}) 
                      </span>
                    </label>
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <textarea
                      aria-invalid="false"
                      class="pf-c-form-control quipucords-form__textarea quipucords-form__textarea-resize-none quipucords-form-control"
                      id="displayScanSources"
                      name="displayScanSources"
                      readonly=""
                      rows="2"
                    >
                       test name 
                    </textarea>
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="scanConcurrency"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                         t(form-dialog.label, {"context":"scan-concurrency"}) 
                      </span>
                    </label>
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <div
                      class="pf-c-number-input "
                    >
                      <div
                        class="pf-c-input-group"
                      >
                        <button
                          aria-disabled="false"
                          aria-label="t(form-dialog.label_touchspin, {\\"
                          class="pf-c-button pf-m-control"
                          context\\":\\"min\\"})"=""
                          data-ouia-component-id="OUIA-Generated-Button-control-1"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe="true"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 448 512"
                            width="1em"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</screen>
`;

exports[`CreateScanDialog Component should render a component, pending: pending 1`] = `
<div
  class="pf-c-modal-box__body"
  id="pf-modal-part-3"
>
  <form
    class="pf-c-form pf-m-horizontal"
    novalidate=""
  >
    <div
      class="pf-c-empty-state quipucords-empty-state"
    >
      <div
        class="pf-c-empty-state__content"
      >
        <span
          aria-hidden="true"
          aria-label="Contents"
          aria-valuetext="Loading..."
          class="pf-c-spinner pf-m-xl pf-c-empty-state__icon"
          role="progressbar"
        >
          <span
            class="pf-c-spinner__clipper"
          >
            <span
              class="pf-c-spinner__lead-ball"
            >
              <span
                class="pf-c-spinner__tail-ball"
              />
              <h3
                class="pf-c-title pf-m-lg"
                data-ouia-component-id="OUIA-Generated-Title-3"
                data-ouia-component-type="PF4/Title"
                data-ouia-safe="true"
              >
                 t(form-dialog.empty-state_title_create-scan, {"context":"pending"}) 
              </h3>
            </span>
          </span>
        </span>
      </div>
    </div>
  </form>
</div>
`;

exports[`CreateScanDialog Component should render nothing if sources are not provided: empty 1`] = `<screen />`;
